### PR TITLE
Set up workflow to clone pymodbus and build a docker image

### PR DESCRIPTION
**Pull request by ch-evergen**
Create a workflow which will pull `pymodbus` source at `v3.1.0` and build a Docker image from it

### CODE REVIEW

File: get-pymodbus.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+git clone -b v3.1.0 https://github.com/pymodbus-dev/pymodbus
```
Comment (serroba) [inline get-pymodbus.sh:3]: could we have done this via `requirements.txt` file?

**Review by serroba — APPROVED** [review]
This works. I would prefer not having to clone manually the external repo, but I don't think we need to spend too much on this either.

File: get-pymodbus.sh (hunk 1 above)
Comment (ch-evergen) [reply get-pymodbus.sh:3]: Hmm I didn't try that, I assumed that the pip installation in a virtual environment would be less suitable for a docker build. I'll give it a go

Comment (ch-evergen) [reply get-pymodbus.sh:3]: no, if we just install pymodbus via pip without the requirements.txt from the source code, it's missing several deps and `pymodbus.server` fails to execute. We would end up recreating most of the requirements.txt in the Dockerfile

**Pull request merged**